runtime/vam/expr/conditional.go

```diff
@@ -103,17 +103,23 @@ func boolMaskRidx(ridx []uint32, bools, errs *roaring.Bitmap, vec vector.Any) {
 		}
 	case *vector.Bool:
 		if ridx != nil {
+			trues := vec
+			if vec.Nulls != nil {
+				// if null and true set to false
+				trues = vector.And(trues, vector.Not(vec.Nulls))
+			}
```

Comment:
You've added this same code to both branches of this if statement. Just add once copy of it before the if.
```suggestion
		trues := vec
		if vec.Nulls != nil {
			// if null and true set to false
			trues = vector.And(trues, vector.Not(vec.Nulls))
		}
		if ridx != nil {
```